src/vulkan/device.cc

```diff
@@ -547,6 +574,26 @@ Result Device::Initialize(
   ptrs_.vkGetPhysicalDeviceMemoryProperties(physical_device_,
                                             &physical_memory_properties_);
 
+  subgroup_size_control_properties_ = {};
+  if (use_physical_device_features_2 &&
+      std::find(
+          required_features.begin(),
+          required_features.end(),
+          kSubgroupSizeControl) != required_features.end()) {
+
+      PFN_vkGetPhysicalDeviceProperties2KHR vkGetPhysicalDeviceProperties2KHR =
```

Comment:
This should use the provided `available_features2` instead of getting the features again.

Reply:
This code is not querying features but device properties.